pupil-spa/src/app/check/check.component.ts

```diff
@@ -122,6 +123,10 @@ export class CheckComponent implements OnInit {
     }
   }
 
+  canDeactivate(): boolean {
+    return false;
```

Comment:
Surely we can deactivate when the checkState is complete?

Reply:
@jon-shipley I have updated this to allow deactivation on warmup intro and check submission view states